ansible/roles/data-pipeline/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+---
+# tasks file for data-pipeline
+- name: Git checkout
+  git:
+    repo: 'https://github.com/Sunbird-Ed/ml-analytics-service.git'
+    dest: "/opt/sparkjobs/ml-analytics-service"
+    version: main
+
+- name: CREATE THE USER
+  user:
+    name: "{{ USER }}"
+    state: present
+    home: "/home/{{ USER }}"
+    shell: /bin/bash
+
+- name: Ensure the locale exists
+  locale_gen:
+    name: en_US.UTF-8
+    state: present
+
+- name: set as default locale
+  command: locale-gen en_US.UTF-8
+
+- name: INSTALL THE PACKGES
+  apt:  
+    update_cache: yes
+    name: "{{ item }}"
+    state: present
+  loop:
+    - openjdk-8-jdk
+    - software-properties-common
+    - python3-pip
+    - python3.6-dev
+    - python3.6-venv
+    - python-virtualenv
+    - zip
+    - unzip
+     
+- name: ADD THE REPOSITORY
+  shell: sudo apt-add-repository universe
+
+- name: CHANGE THE OWNERSHIP FOR THIS {{ BASEPATH }} DIRECTORY
+  file:
+    path: "{{ BASEPATH }}"
+    mode: "0777"
+    recurse: yes
+    owner: "{{ USER }}"
+    group: "{{ USER }}"
+
+- name: CREATE THE DIRECTORY
+  file:
+    path: "{{ WORKDIR }}"
+    state: directory
+    owner: "{{ USER }}"
+    group: "{{ USER }}"
+
+- name: Change user and create working directory under opt dir and install python virtual environment
+  shell: "cd {{ WORKDIR }} && virtualenv --python=python3.6 spark_venv"
+  become: yes
+  become_user: "{{ USER }}"
+
+- name: Create necessary logs folders for pipeline
+  become_user: "{{ USER }}"
+  become: yes
+  shell: "cd {{ WORKDIR }} && mkdir -p logs/observation/status logs/observation_evidence logs/project logs/survey logs/survey_evidence"
+
+- name: Initiate virtualenv
+  pip:
+    virtualenv: "{{WORKDIR }}/spark_venv"
+    virtualenv_python: python3.6
+    requirements: "{{ WORKDIR }}/ml-analytics-service/requirements.txt"
+
+- name: create the directory
+  file:
+    path:  "{{ WORKDIR }}/faust_as_service"
+    state: directory
+    owner: "{{ USER }}"
+    group: "{{ USER }}"
+
+- name: Creating a faust service shell file in executable mode
+  copy:
+    dest: "{{ WORKDIR }}/faust_as_service/faust.sh"
+    owner: "{{ USER }}"
+    group: "{{ USER }}"	
+    mode: 0777
+    content: |
```

Comment:
keep all these in files/ directory, and copy it over to desination. Use loop and single block

Reply:
for each file the copying content is different, even though we iterate file in the single loop 
we will not be able to copy the content

Reply:
we don't care about content. That's why we're using different files. Add this to the fil copy module.